pytest_fixtures/broker.py

```diff
@@ -1,12 +1,43 @@
 import pytest
 from broker.broker import VMBroker
+from wait_for import wait_for
 
 from robottelo.constants import BROKER_RHEL77
 from robottelo.hosts import Capsule
 from robottelo.hosts import ContentHost
 from robottelo.hosts import Satellite
 
 
+@pytest.fixture(scope="session")
+def satellite_factory():
+    def factory(retry_limit=3, delay=300, **broker_args):
+        vmb = VMBroker(host_classes={'host': Satellite}, workflow='deploy-sat-lite', **broker_args)
+        timeout = (1200 + delay) * retry_limit
+        sat = wait_for(
+            vmb.checkout, func_kwargs=broker_args, timeout=timeout, delay=delay, fail_condition=[]
+        )
+        return sat.out
+
+    return factory
+
+
+@pytest.fixture(scope="session")
+def capsule_factory():
+    def factory(retry_limit=3, sleep=300, **broker_args):
+        vmb = VMBroker(
+            host_classes={'host': Capsule}, workflow='deploy-sat-capsule', **broker_args
+        )
+        retries = 0
+        while not (cap := vmb.checkout()) and (retries < retry_limit):
```

Comment:
I'd use same wait_for logic in both factory functions here.

Given the similarity, maybe this can be a single function with the first arg being `Satellite | Capsule`?

Reply:
YOU FOUND THE "HIDDEN" REQUIRED CHANGE!!!!!!!

Reply:
fixed

Reply:
See I pay attention!